jvm/src/main/scala/io/glutenproject/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -507,10 +543,41 @@ case class UnionExecTransformer(children: Seq[SparkPlan]) extends SparkPlan with
     throw new UnsupportedOperationException(s"This operator doesn't support getChild.")
   }
 
-  override def doValidate(): Boolean = false
+  override def doValidate(): Boolean = {
+    val substraitContext = new SubstraitContext
+    val operatorId = substraitContext.nextOperatorId
+    val relNode = try {
+      val attrList = new util.ArrayList[Attribute]()
+      for (attr <- children(0).output)
+      {
+        attrList.add(attr)
+      }
+      RelBuilder.makeReadRel(attrList, substraitContext, operatorId)
+    } catch {
+      case e: Throwable =>
+        logDebug(s"Validation failed for ${this.getClass.toString} due to ${e.getMessage}")
+        return false
+    }
+    if (relNode != null && GlutenConfig.getConf.enableNativeValidation) {
+      val planNode = PlanBuilder.makePlan(substraitContext, Lists.newArrayList(relNode))
+      val validator = new ExpressionEvaluator()
+      validator.doValidate(planNode.toProtobuf.toByteArray)
+    } else {
+      true
+    }
+  }
+
 
   override def doTransform(context: SubstraitContext): TransformContext = {
-    throw new UnsupportedOperationException(s"This operator doesn't support doTransform.")
+    val operatorId = context.nextOperatorId
+    val attrList = new util.ArrayList[Attribute]()
+    for (attr <- children(0).output)
```

Comment:
How about adding some check for whether children is empty here?

Reply:
add check in `doValidate()`